src/scripts.c

```diff
@@ -51,7 +51,11 @@ static void update_triggers(void)
 {
 	__attribute__((unused)) int ret = 0;
 
-	ret = system("/usr/bin/systemctl --no-block daemon-reload");
+	if (need_systemd_reexec) {
+		ret = system("/usr/bin/systemctl daemon-reexec");
+	}
```

Comment:
Since `need_systemd_reexec` will generally be false, maybe we could optimize for that case? Like so:

```
if (need_systemd_reexec) {
        ret = system("/usr/bin/systemctl daemon-reexec");
        ret = system("/usr/bin/systemctl daemon-reload");
} else {
        ret = system("/usr/bin/systemctl --no-block daemon-reload");
}
```